curator-recipes/src/test/java/org/apache/curator/framework/recipes/nodes/TestPersistentTtlNode.java

```diff
@@ -187,4 +187,34 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) th
             assertNull(client.checkExists().forPath("/test"));
         }
     }
+
+    @Test
+    public void testTouchNodeNotCreated() throws Exception {
+        try (CuratorFramework client =
+                CuratorFrameworkFactory.newClient(server.getConnectString(), new RetryOneTime(1))) {
+            client.start();
+            final long ttlMs = 1_000L;
+            try (PersistentTtlNode node = new PersistentTtlNode(client, "/test", ttlMs, new byte[0])) {
+                node.start();
+                assertTrue(node.waitForInitialCreate(timing.session(), TimeUnit.MILLISECONDS));
+                // Give some minor time for touch node to be created. Will worked after patch
+                for (int i = 1; i <= 5; i++) {
+                    if (client.checkExists().forPath("/test") != null) {
+                        break;
+                    }
+                    Thread.sleep(10L);
+                }
+            }
+        }
+        try (CuratorFramework client1 =
+                CuratorFrameworkFactory.newClient(server.getConnectString(), new RetryOneTime(1))) {
+            client1.start();
+            assertTrue(client1.blockUntilConnected(2, TimeUnit.SECONDS));
+            Thread.sleep(3_000L);
+            assertNull(client1.checkExists().forPath("/test/touch"));
```

Comment:
We could move this before `sleep` to ensure that the "touch" node is not created.